hydra_agent/querying_mechanism.py

```diff
@@ -526,110 +539,132 @@ def user_query(self, query):
-                logger.info("Error: Incorrect query")
-                return None
-            else:
-                data = self.properties.get_members_properties(query)
-                return data
-        elif "object" in query:
-            if query[-1] == " ":
-                logger.info("Error: incorrect query")
-                return None
-            check_query = self.check_fine_query(query)
-            if isinstance (check_query, RequestError):
-                logger.info("Error: Incorrect query")
-                return None
-            else:
-                data = self.properties.get_object_property(query)
-                return data
-        elif "Collection" in query:
-            if query[-1] == " ":
-                logger.info("Error: incorrect query")
-                return None
-            check_query = self.check_fine_query(query)
-            if isinstance (check_query, RequestError):
-                logger.info("Error: Incorrect query")
-                return None
             else:
-                data = self.properties.get_collection_properties(query)
-                return data
-        elif "class" in query and "property_value" in query:
-            check_query = self.check_fine_query(query)
-            if isinstance (check_query, RequestError):
-                logger.info("Error: Incorrect query")
+                if " and " in query or " or " in query:
+                    if query[-1] == " " or query[-3] == "and" or query[-2] == "or":
+                        logger.info("Error: incorrect query")
+                        return None
+                    query_len = len(query.split())
+                    and_or_count = query.count("and") + query.count("or")
+                    if query_len != (and_or_count + 2 * (and_or_count + 1)):
+                        logger.info("Error: Incorrect query")
+                        return None
+                    data = self.compare.object_property_comparison_list(query)
+                    return data
+                elif query.count(" ") == 1:
+                    key, value = query.split(" ")
+                    print("query: ", query)
+                    search_index = "fs:" + key + ":" + value
+                    for key in self.connection.keys():
+                        if search_index == key.decode("utf8"):
+                            data = self.connection.smembers(key)
+                            return data
+                logger.info("Incorrect query: Use 'help' to know about\
+                            querying format")
                 return None
+        elif "create" in query:
+            query = query.replace("create", "").strip()
+            if "in" in query:
+                query = query.replace("in", "").strip()
+                collection = query.split(" ", 1)[0]
+                new_object = query.replace(collection, "").strip()
+                self.members = CollectionmembersQuery(self.api_doc,
```

Comment:
Not sure you need to use `self.members` here, each query is run one time so using a simple variable `members` will do.